cocotb/share/lib/embed/gpi_embed.c

```diff
@@ -51,6 +54,68 @@ static char *argv[] = { progname };
 
 static PyObject *pEventFn = NULL;
 
+
+/**
+ * @name    handle_venv
+ * @brief   Set 
+ * @ingroup python_c_api
+ *
+ * GILState before calling: N/A
+ *
+ * GILState after calling: N/A
+ *
+ * Checks if running in Python virtualenv and if so,
+ * calls Py_SetProgramName with correct path to interpreter
+ */
+
+void handle_venv(void)
+{
+    static char venv_path[PATH_MAX];
+#if PY_MAJOR_VERSION >= 3
+    static wchar_t venv_path_w[PATH_MAX];
+#endif
+
+    char *venv_path_home = getenv("VIRTUAL_ENV");
+    if (venv_path_home) {
+        strncpy(venv_path, venv_path_home, sizeof(venv_path));
+        if (venv_path[sizeof(venv_path) - 1])
+            goto venv_err;
+
+#if defined(_WIN32)
+#if defined(__MINGW32__) || defined (__CYGWIN32__)
+        strncat(venv_path, "/Scripts/python", sizeof(venv_path) - strlen(venv_path) - 1);
+#else
+        strncat(venv_path, "\\Scripts\\python", sizeof(venv_path) - strlen(venv_path) - 1);
+#endif
+#else
+        strncat(venv_path, "/bin/python", sizeof(venv_path) - strlen(venv_path) - 1);
+#endif
```

Comment:
You can move all those defines out of this function or at least avoid duplicating the `strncat()` code by doing something like

```c
#if defined(_WIN32)
#  if defined(__MINGW32__) || defined (__CYGWIN32__)
const char* PYTHON_INTERPRETER_PATH = "/Scripts/python";
#  else
const char* PYTHON_INTERPRETER_PATH = "\\Scripts\\python";
#  endif
#else
const char* PYTHON_INTERPRETER_PATH = "/bin/python";
#endif

You can use defines instead of const strings if you want. This code would go well into the beginning of the file.